image_publisher/src/nodelet/image_publisher_nodelet.cpp

```diff
@@ -0,0 +1,194 @@
+*     contributors may be used to endorse or promote products derived
+*     from this software without specific prior written permission.
+* 
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+*  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+*  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+*  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+*  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+*  POSSIBILITY OF SUCH DAMAGE.
+*********************************************************************/
+
+#include <ros/ros.h>
+#include <nodelet/nodelet.h>
+#include <cv_bridge/cv_bridge.h>
+#include <image_publisher/ImagePublisherConfig.h>
+#include <image_transport/image_transport.h>
+#include <sensor_msgs/CameraInfo.h>
+#include <camera_info_manager/camera_info_manager.h>
+#include <opencv2/highgui/highgui.hpp>
+#include <dynamic_reconfigure/server.h>
+#include <boost/assign.hpp>
+using namespace boost::assign;
+
+namespace image_publisher {
+class ImagePublisherNodelet : public nodelet::Nodelet
+{
+  dynamic_reconfigure::Server<image_publisher::ImagePublisherConfig> srv;
+
+  image_transport::CameraPublisher pub_;
+
+  boost::shared_ptr<image_transport::ImageTransport> it_;
+  ros::NodeHandle nh_;
+
+  cv::VideoCapture cap_;
+  cv::Mat image_;
+  int subscriber_count_;
+  ros::Timer timer_;
+
+  std::string frame_id_;
+  std::string filename_;
+  bool flip_image_;
+  int flip_value_;
+  sensor_msgs::CameraInfo camera_info_;
+  
+
+  void reconfigureCallback(image_publisher::ImagePublisherConfig &new_config, uint32_t level)
+  {
+    frame_id_ = new_config.frame_id;
+
+    timer_ = nh_.createTimer(ros::Duration(1.0/new_config.publish_rate), &ImagePublisherNodelet::do_work, this);
+
+    camera_info_manager::CameraInfoManager c(nh_);
+    if ( !new_config.camera_info_url.empty() ) {
+      try {
+        c.validateURL(new_config.camera_info_url);
+        c.loadCameraInfo(new_config.camera_info_url);
+        camera_info_ = c.getCameraInfo();
+      } catch(cv::Exception &e) {
+        NODELET_ERROR("camera calibration failed to load: %s %s %s %i", e.err.c_str(), e.func.c_str(), e.file.c_str(), e.line);
+      }
+    }
+    
+
+  }
+
+  void do_work(const ros::TimerEvent& event)
+  {
+    // Transform the image.
+    try
+    {
+      if ( cap_.isOpened() ) {
+        if ( ! cap_.read(image_) ) {
+          cap_.set(CV_CAP_PROP_POS_FRAMES, 0);
+        }
+      }
+      if (flip_image_)
+        cv::flip(image_, image_, flip_value_);
```

Comment:
@wkentaro I commented it would be useful to have here: https://github.com/ros-perception/image_pipeline/pull/173#issuecomment-170364563 I personally have used this feature a few times already (because of mounting on pourpose, or not, cameras flipped). I also have a image_publisher based on opencv like this one, so I wouldn't like to lose this ability (altho I would like to migrate to this publisher).